codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -87,18 +87,18 @@ public void remoteSettings(Http2Settings settings) throws Http2Exception {
             connection.remote().allowPushTo(pushEnabled);
         }
 
-        Long maxConcurrentStreams = settings.maxConcurrentStreams();
+        Integer maxConcurrentStreams = settings.maxConcurrentStreams();
         if (maxConcurrentStreams != null) {
             // TODO(scott): define an extension setting so we can communicate/enforce the maxStreams limit locally.
-            connection.local().maxStreams((int) min(maxConcurrentStreams, MAX_VALUE), MAX_VALUE);
+            connection.local().maxStreams(maxConcurrentStreams, MAX_VALUE);
         }
 
         Long headerTableSize = settings.headerTableSize();
         if (headerTableSize != null) {
             outboundHeaderTable.maxHeaderTableSize((int) min(headerTableSize, MAX_VALUE));
         }
 
-        Integer maxHeaderListSize = settings.maxHeaderListSize();
+        Long maxHeaderListSize = settings.maxHeaderListSize();
```

Comment:
To be spec compliant settings are generally treated as unsigned 32 bit numbers (https://tools.ietf.org/html/rfc7540#section-6.5.1) (unless other parts of the spec mean that doesn't make sense ... aka `MAX_CONCURRENT_STREAMS`)